catroid/src/main/java/org/catrobat/catroid/ui/recyclerview/controller/ScriptController.kt

```diff
@@ -162,8 +164,15 @@ class ScriptController {
         destinationSprite.scriptList.add(script)
     }
 
-    @Throws(CloneNotSupportedException::class)
+    @Throws(CloneNotSupportedException::class, UnpackUserDefinedBrickException::class)
     fun unpack(scriptToUnpack: Script, destinationSprite: Sprite) {
+        if (scriptToUnpack is UserDefinedScript) {
+            for (script: Script in destinationSprite.scriptList) {
+                if (script is UserDefinedScript && script.userDefinedBrickID == scriptToUnpack.userDefinedBrickID) {
```

Comment:
I am sorry to jump in here, but I was just curious about this ticket. Please do not compare `UUID`s with `==` as this just checks for reference equality. Use `.equals()` to check for the same **value** instead.

Reply:
The == should be the correct way to handle this because it is written in kotlin. Referential equality in kotlin is ===. As far as I know the == in kotlin basically boils down to a?.equals(b) ?: (b === null).

Reply:
You are perfectly right! It just would have been a problem when written in Java. Sorry for the inconvenience.